### PR TITLE
Fixed tile scale computation (#6912)

### PR DESCRIPTION
**Pull request by mikitex70**
Fixed tile scale computation when the `UNITS` directive is used in the map (for example `UNITS dd`).
Added some test  to verify that the scale are computed correctly.

### CODE REVIEW

File: msautotest/misc/tiles_scales_z2.map [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  LAYER
+    DEBUG 5
+    NAME "canada-poly"
+    TYPE POLYGON
+    STATUS DEFAULT
+    CONNECTIONTYPE OGR
+    CONNECTION "data/canada.dgn"
+    DATA "elements"
+    MINSCALE 30000000
```
Comment (geographika) [inline msautotest/misc/tiles_scales_z2.map:30]: `MINSCALE` and `MAXSCALE` are deprecated. These should be replaced with `MINSCALEDENOM` and `MAXSCALEDENOM`.

File: msautotest/misc/tiles_scales_z3.map [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  LAYER
+    DEBUG 5
+    NAME "canada-poly"
+    TYPE POLYGON
+    STATUS DEFAULT
+    CONNECTIONTYPE OGR
+    CONNECTION "data/canada.dgn"
+    DATA "elements"
+    MINSCALE 30000000
+    MAXSCALE 70000000
```
Comment (geographika) [inline msautotest/misc/tiles_scales_z3.map:31]: As above

File: msautotest/misc/tiles_scales_z4.map [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  LAYER
+    DEBUG 5
+    NAME "canada-poly"
+    TYPE POLYGON
+    STATUS DEFAULT
+    CONNECTIONTYPE OGR
+    CONNECTION "data/canada.dgn"
+    DATA "elements"
+    MINSCALE 30000000
+    MAXSCALE 70000000
```
Comment (geographika) [inline msautotest/misc/tiles_scales_z4.map:31]: As above

**Review by geographika — CHANGES_REQUESTED** [review]

**Comment by geographika** [comment]
Thanks @mikitex70 for this, and for adding in the tests. 

For the tests could you add in 2 new images files (even if they are duplicates of `tile_scale_empty.png`) with names that match the test Mapfile? This is to ensure if a test fails it doesn't pick up a previously created file of the same name (see discussions in #6919).

**Review by geographika — APPROVED** [review]

**Comment by jmckenna** [comment]
thanks @mikitex70 !

**Pull request merged**